### PR TITLE
adds API tests to verify requestedName is used during ca auto enroll

### PR DESCRIPTION
**Pull request by andrewpmartinez**

### CODE REVIEW

File: tests/enrollment_ca_auto_test.go [hunk 1]
```diff
@@ -128,11 +132,12 @@ func Test_IdentityEnrollment(t *testing.T) {
 			})
 
 			t.Run("can enroll with a name", func(t *testing.T) {
-				t.Run("can enroll without a name and empty JSON object", func(t *testing.T) {
+				t.Run("can enroll with", func(t *testing.T) {
```
Comment (dovholuknf) [inline tests/enrollment_ca_auto_test.go:135]: can enroll with 'what'?

Comment (dovholuknf) [reply tests/enrollment_ca_auto_test.go:135]: nevermind i see it referenced below now :)

File: tests/enrollment_ca_auto_test.go [hunk 2]
```diff
@@ -143,12 +148,35 @@ func Test_IdentityEnrollment(t *testing.T) {
...
 						SetHeader("content-type", "application/json").
-						SetBody(`{"name": "` + eid.New() + `"}`).
+						SetBody(`{"name": "` + name + `"}`).
 						Post("enroll?method=ca")
 
 					ctx.Req.NoError(err)
 
 					ctx.Req.Equal(http.StatusOK, resp.StatusCode())
+
+					t.Run("the name was given to the identity", func(t *testing.T) {
```
Comment (dovholuknf) [inline tests/enrollment_ca_auto_test.go:158]: nit: the name given to the identity or the name that was given to the identity

File: tests/enrollment_ca_auto_test.go [hunk 3]
```diff
@@ -143,12 +148,35 @@ func Test_IdentityEnrollment(t *testing.T) {
...
+						ctx.Req.NoError(err)
+						standardJsonResponseTests(resp, http.StatusOK, t)
+
+						envelope := rest_model.ListIdentitiesEnvelope{}
+
+						err = envelope.UnmarshalBinary(resp.Body())
+						ctx.Req.NoError(err)
+						found := false
+						for _, identity := range envelope.Data {
+							if *identity.Name == name {
```
Comment (dovholuknf) [inline tests/enrollment_ca_auto_test.go:172]: what happens when the same cased name is provided?

**Review by dovholuknf — APPROVED** [review]

File: tests/enrollment_ca_auto_test.go (hunk 3 above)
Comment (andrewpmartinez) [reply tests/enrollment_ca_auto_test.go:172]: This is testing the ability for `[requestedName]`, which is provided during enrollment, is passed through to the controller and applied when required by the name formatting. The value passed in is used opaquely - so whatever the casing is, it is used.

**Pull request merged**